src/segmantic/seg/evaluation.py

```diff
@@ -0,0 +1,242 @@
+        if l == 0:
+            return (0, 0, 0)
+        hue = l / (2.0 * max_label) + (l % 2) * 0.5
+        hue = min(hue, 1.0)
+        return colorsys.hls_to_rgb(hue, 0.5, 1.0)
+
+    col = np.zeros((num_classes, 3))
+
+    random.seed(0)
+    for i in random.sample(range(num_classes), num_classes):
+        r, g, b = random_color(i, num_classes)
+        col[i, 0] = r
+        col[i, 1] = g
+        col[i, 2] = b
+    return colors.ListedColormap(col)
+
+
+def compute_confusion(
+    num_classes: int, y_pred: np.ndarray, y: np.ndarray
+) -> np.ndarray:
+    """Compute confusion matrix similar to sklearn.metrics.confusion_matrix
+
+    Args:
+        num_classes (int): number of labels including '0', i.e. max(y)+1
+        y_pred (np.ndarray): predicted labels
+        y (np.ndarray): true labels
+
+    Returns:
+        np.ndarray: [description]
+    """
+    try:
+        from numba import njit
+
+        @njit
+        def _compute_confusion(num_classes: int, y_pred: np.ndarray, y: np.ndarray):
+            cm = np.zeros((num_classes, num_classes))
+            for i in range(num_classes):
+                cm[y[i], y_pred[i]] += 1
+            return cm
+
+        return _compute_confusion(num_classes, y_pred, y)
+    except:
+        # fall back to naive approach
+        cm = np.zeros((num_classes, num_classes))
+        for t, p in zip(y_pred, y):
+            cm[t, p] += 1
+    return cm
+
+
+def plot_confusion_matrix(
+    cm: np.ndarray,
+    target_names: List[str],
+    title: str = "Confusion matrix",
+    cmap=None,
+    normalize: bool = True,
+    file_name: Optional[Path] = None,
+) -> None:
+    """
+    given a sklearn confusion matrix (cm), make a nice plot
+
+    Arguments
+    ---------
+    cm:           confusion matrix from sklearn.metrics.confusion_matrix
+
+    target_names: given classification classes such as [0, 1, 2]
+                  the class names, for example: ['high', 'medium', 'low']
+
+    title:        the text to display at the top of the matrix
+
+    cmap:         the gradient of the values displayed from matplotlib.pyplot.cm
+                  see http://matplotlib.org/examples/color/colormaps_reference.html
+                  plt.get_cmap('jet') or plt.cm.Blues
+
+    normalize:    If False, plot the raw numbers
+                  If True, plot the proportions
+
+    Usage
+    -----
+    plot_confusion_matrix(cm           = cm,                  # confusion matrix created by
+                                                              # sklearn.metrics.confusion_matrix
+                          normalize    = True,                # show proportions
+                          target_names = y_labels_vals,       # list of names of the classes
+                          title        = best_estimator_name) # title of graph
+
+    Citiation
+    ---------
+    http://scikit-learn.org/stable/auto_examples/model_selection/plot_confusion_matrix.html
+
+    """
+    import matplotlib.pyplot as plt
```

Comment:
why import inside? would it not be better to separate plotting from computation modules?

Reply:
agreed, I will move it to a different module

Reply:
done